src/include/miopen/generic_search.hpp

```diff
@@ -335,45 +389,54 @@ auto GenericSearch(const Solver s, const Context& context_, const AnyInvokeParam
     HeartBeat<PerformanceConfig> heartbeat;
     heartbeat.Start();
 
-    if(!miopen::IsCacheDisabled()) // Otherwise precompilation is useless.
+    const auto total_threads = GetTuningThreadsMax();
+    ThreadSafeQueue<std::tuple<PerformanceConfig, ConvSolution, bool>> solution_queue;
```

Comment:
It may be worth using `boost::lockfree::queue`

Reply:
I would prefer not to add additional dependency on boost, it is something we are striving to get rid of.